streams/src/test/java/org/apache/kafka/streams/state/internals/SessionBytesStoreTest.java

```diff
@@ -0,0 +1,553 @@
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.kafka.clients.producer.MockProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.errors.DefaultProductionExceptionHandler;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.SessionWindow;
+import org.apache.kafka.streams.processor.internals.MockStreamsMetrics;
+import org.apache.kafka.streams.processor.internals.RecordCollector;
+import org.apache.kafka.streams.processor.internals.RecordCollectorImpl;
+import org.apache.kafka.streams.processor.internals.testutil.LogCaptureAppender;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.SessionStore;
+import org.apache.kafka.test.InternalMockProcessorContext;
+import org.apache.kafka.test.TestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public abstract class SessionBytesStoreTest {
+
+    protected static final long SEGMENT_INTERVAL = 60_000L;
+    protected static final long RETENTION_PERIOD = 10_000L;
+
+    protected SessionStore<String, Long> sessionStore;
+    protected InternalMockProcessorContext context;
+
+    private final List<KeyValue<byte[], byte[]>> changeLog = new ArrayList<>();
+
+    private final Producer<byte[], byte[]> producer = new MockProducer<>(true,
+        Serdes.ByteArray().serializer(),
+        Serdes.ByteArray().serializer());
+
+    abstract <K, V> SessionStore<K, V> buildSessionStore(final long retentionPeriod,
+                                                          final Serde<K> keySerde,
+                                                          final Serde<V> valueSerde);
+
+    abstract String getMetricsScope();
+
+    abstract void setClassLoggerToDebug();
```

Comment:
Why on WindowBytesStoreTest we need a flag whereas here we do not? Both used a segmented store internally right?

Reply:
Ah, good catch. Actually the flag should have been removed as no logging is done on the RocksDBXXXStore layer -- it's all in AbstractRocksDBSegmentedBytesStore. 

Weird that shouldNotThrowInvalidRangeExceptionWithNegativeFromKey() passed anyway